sdk/src/builder.rs

```diff
@@ -972,6 +976,65 @@ impl Builder {
         }
     }
 
+    #[cfg(feature = "file_io")]
+    // Internal utiltiy to set format and title based on destination filename.
+    //
+    // Also sets the instance_id to a new UUID and ensures the destination file does not exist.
+    fn set_asset_from_dest<P: AsRef<Path>>(&mut self, dest: P) -> Result<()> {
+        let path = dest.as_ref();
+        if !path.exists() {
+            // ensure the path to the file exists
+            if let Some(output_dir) = path.parent() {
+                std::fs::create_dir_all(output_dir)?;
+            }
+        } else {
+            // if the file exists, we need to remove it to avoid appending to it
+            return Err(crate::Error::BadParam(
+                "Destination file already exists".to_string(),
+            ));
+        };
+        self.definition.format =
+            crate::format_from_path(path).ok_or(crate::Error::UnsupportedType)?;
+        self.definition.instance_id = format!("xmp:iid:{}", Uuid::new_v4());
+        if self.definition.title.is_none() {
+            if let Some(title) = path.file_name() {
+                self.definition.title = Some(title.to_string_lossy().to_string());
+            }
+        }
+        Ok(())
+    }
+
+    /// Sign a set of fragmented BMFF files for DASH streaming.
```

Comment:
You can remove the data comment.  It should be BMFF based.  So HLS for example should work as well.

Reply:
We should add a note here that the code does not work if the files already have C2PA data.  That is a known limitation.  I will add a story to support that.